sycl/source/detail/scheduler/scheduler.cpp

```diff
@@ -70,6 +70,7 @@ EventImplPtr Scheduler::addCG(std::unique_ptr<detail::CG> CommandGroup,
                               QueueImplPtr Queue) {
   EventImplPtr NewEvent = nullptr;
   const bool IsKernel = CommandGroup->getType() == CG::KERNEL;
+  const bool IsHostKernel = CommandGroup->getType() == CG::RUN_ON_HOST_INTEL;
```

Comment:
I would like some `std::variant` here when C++17 lands instead of all this manual pattern matching.